src/Main.tsx

```diff
@@ -2,22 +2,154 @@ import React from 'react';
 import MatchSelector from './components/MatchSelector';
 import StandingsTable from './components/StandingsTable';
 import './Main.css';
+// import { Players } from './types/index'
 
 
-class Main extends React.Component {
+interface MainState {
+    regionToDisplay: string;
+    groupToDisplay: string;
+    players: string[]
+    matches: string[][];
+    results: string;
+}
+
+class Main extends React.Component<{},MainState> {
+    constructor(props: {}) {
+        super(props);
+        const defaultRegion = "APAC";
+        const defaultGroup = "A";
+        const players = selectPlayers(defaultRegion, defaultGroup);
+        this.state = {
+            regionToDisplay: defaultRegion,
+            groupToDisplay: defaultGroup,
+            players: players,
+            matches: selectMatches(players),
+            results: ""
+        }
+    }
+
+    handleClickRegion(region: string) {
+        if (this.state.regionToDisplay !== region) {
+            const players = selectPlayers(region, 'A');
+            this.setState({
+                // reset view
+                groupToDisplay: 'A',
+                results: "",
+                // update info
+                regionToDisplay: region,
+                players: players,
+                matches: selectMatches(players),
+
+            });
+        }
+    }
+
+    handleClickGroup(group: string) {
+        if (this.state.groupToDisplay !== group) {
+            const players = selectPlayers(this.state.regionToDisplay, group);
+            this.setState({ 
+                // reset view
+                results: "",
+                // update info
+                groupToDisplay: group ,
+                players: players,
+                matches: selectMatches(players),
+            });
+        }
+    }
+
+    handleClickMatch(mIndex: number, wIndex: number) {
+        this.setState({
```

Comment:
tu peux destructurer les objets en js afin de ne pas répeter tout le temps les mêmes trucs:
`const { result, matches} = this.state` => cela est équivalent à 
```
const result = this.state.result;
const matches = this.state.matches;
```